### PR TITLE
Add ErrorDetail model

### PR DESCRIPTION
**Pull request by YalinLi0312**

### CODE REVIEW

File: specification/appconfiguration/AppConfiguration/models.tsp [hunk 1]
```diff
@@ -151,6 +151,36 @@ model Error {
...
+  code?: string;
+
+  @doc("An object containing more specific information than the current object about the error.")
+  #suppress "@azure-tools/typespec-azure-core/property-name-conflict" "Property 'innererror' having the same name as its enclosing model will cause problems with C# code generation."
+  innererror?: InnerError;
+}
+
+@doc("The details of an error.")
+@error
+model ErrorDetail {
```
Comment (catalinaperalta) [inline specification/appconfiguration/AppConfiguration/models.tsp:167]: why cant we just do `model ErrorDetail is Foundations.Error` ?

Comment (YalinLi0312) [reply specification/appconfiguration/AppConfiguration/models.tsp:167]: Got this error when compile: 
image

Comment (catalinaperalta) [reply specification/appconfiguration/AppConfiguration/models.tsp:167]: These errors look like you tried to define the model inline on the property which is not supported syntax.

Comment (catalinaperalta) [reply specification/appconfiguration/AppConfiguration/models.tsp:167]: you just need to define the model independently and then refer to it in the property field

Comment (catalinaperalta) [reply specification/appconfiguration/AppConfiguration/models.tsp:167]: If using `model ErrorDetails is Foundations.Error` is too complicated to get to work with the emitters, then let's make sure the models you add to the spec match the service's rest api.

**Review by XiaofeiCao — APPROVED** [review]

**Comment by XiaofeiCao** [comment]
Looks good from Java SDK perspective.

File: specification/appconfiguration/AppConfiguration/models.tsp (hunk 1 above)
Comment (YalinLi0312) [reply specification/appconfiguration/AppConfiguration/models.tsp:167]: The models are consisting with what we are using now. We can use `Foundations.Error` when emitter supports renaming on it.

**Review by lmazuel — CHANGES_REQUESTED** [review]
Let's talk offline first, this shouldn'b necessary.

**Comment by YalinLi0312** [comment]
Closed it as we all languages are good to use the model from azure core.

**Pull request closed without merging**